data/campaigns/Descent_Into_Darkness/units/Mal_Keshar.cfg

```diff
@@ -0,0 +1,652 @@
+        [/effect]
+    ) {VARIANT_NAME}}
+#enddef
+
+#define ANCIENT_LICH_RESIST VARIANT_NAME
+    {MAL_KESHAR_ADVANCEMENT 0 80 (
+        id=ancient_lich_resist
+        description= _ "Gains +10% resistance to blade, impact, and pierce"
+        image=icons/potion_red_medium.png
+        [effect]
+            apply_to=resistance
+            [resistance]
+                blade=-10
+                impact=-10
+                pierce=-10
+            [/resistance]
+        [/effect]
+    ) {VARIANT_NAME}}
+#enddef
+
+#define ANCIENT_LICH_NOVA VARIANT_NAME
+    {MAL_KESHAR_ADVANCEMENT 2 0 (
+        id=ancient_lich_nova
+        description= _ "Chill tempest gains area of effect damage"
+        image=attacks/iceball.png
+        [effect]
+            apply_to=attack
+            name=chill tempest
+            [set_specials]
+                {WEAPON_SPECIAL_MAGICAL}
+                {WEAPON_SPECIAL_FROST_NOVA}
+            [/set_specials]
+        [/effect]
+        [effect]
+            apply_to=max_experience
+            increase=-100%
+        [/effect]
+        [effect]
+            apply_to=max_experience
+            increase=119
+        [/effect]
+    ) {VARIANT_NAME}}
+#enddef
+
+#define ANCIENT_LICH_SPELLBIND VARIANT_NAME
+    {MAL_KESHAR_ADVANCEMENT 4 0 (
+        id=ancient_lich_spellbind
+        description= _ "Shadow wave reduces incoming magic damage"
+        image=attacks/dark-missile.png
+        [effect]
+            apply_to=attack
+            name=shadow wave
+            [set_specials]
+                {WEAPON_SPECIAL_MAGICAL}
+                {WEAPON_SPECIAL_SPELLBIND}
+            [/set_specials]
+        [/effect]
+        [effect]
+            apply_to=attack
+            name=shadow wave
+            increase_damage=1
+        [/effect]
+        [effect]
+            apply_to=max_experience
+            increase=-100%
+        [/effect]
+        [effect]
+            apply_to=max_experience
+            increase=119
+        [/effect]
+    ) {VARIANT_NAME}}
+#enddef
+
+#define ANCIENT_LICH_REND VARIANT_NAME
+    {MAL_KESHAR_ADVANCEMENT 5 0 (
+        id=ancient_lich_blood
+        description= _ "Drain HP from nearby enemies, but lose HP every turn"
+        image=attacks/curse.png
+        [effect]
+            apply_to=new_ability
+            [abilities]
+                {ABILITY_SOUL_REND}
+            [/abilities]
+        [/effect]
+        [effect]
+            apply_to=attack
+            name=touch
+            [set_specials]
+                {WEAPON_SPECIAL_REAP}
+            [/set_specials]
+        [/effect]
+        [effect]
+            apply_to=attack
+            name=touch
+            increase_damage=1
+        [/effect]
+        [effect]
+            apply_to=max_experience
+            increase=-100%
+        [/effect]
+        [effect]
+            apply_to=max_experience
+            increase=119
+        [/effect]
+    ) {VARIANT_NAME}}
+#enddef
+
+#define MAL_KESHAR_LICH_BASE
```

Comment:
Would it make sense to use [base_unit] since I assume most of these properties are just copies from the Lich unit type?

Reply:
I can't figure out how to make it work with the ancient lich macro, since I have to overwrite the animations or it'll inherit the lich ones. If someone else can figure out how to use base_unit with that, I can put it in.